### ISSUE TITLE
Add end to end tests for Parca with Agent

### ISSUE THREAD

**Issue by kakkoyun** [parca-dev/parca#353]
* Spawn up a cluster on CI
* Deploy
* Make sure basic functionality actually works

**Comment by knrt10** [comment]
One way I think of it is

- Create a pipeline and steps to install minikube and then create a cluster
- Once the cluster is up on a different steps deploy parca on the[ k8s cluster](https://www.parca.dev/docs/kubernetes)
- Then to test the functionality I would say checking whether the pods are up and running or not. And whether the metrics endpoint has some data popullated.

If this makes sense, I can start working on it.

**Comment by kakkoyun** [comment]
@knrt10 Yes, this is exactly what we're working on, to be more specific, @Sylfrena is working on. 

Feel free to contribute the improvements after she's done with the initial work.
